### PR TITLE
Adding storage class pd-ssd to GCP clusters

### PR DESCRIPTION
**Pull request by aliceh**
For test purposes made pd-ssd the default storage class.

### CODE REVIEW

File: deploy/osd-storage/cluster-storage-gcp-standard.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: standard
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "false"
+provisioner: kubernetes.io/gce-pd
+parameters:
+  type: pd-standard
```
Comment (dustman9000) [inline deploy/osd-storage/cluster-storage-gcp-standard.yaml:9]:
```suggestion
  type: pd-ssd
```
We already have pd-standard, we want to add pd-ssd right?

File: deploy/osd-storage/config.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+deploymentMode: SelectorSyncSet
+selectorSyncSet: 
+  matchExpressions: 
+    - key: hive.openshift.io/cluster-platform
+      operator: In
+      values: [gcp]
```
Comment (fahlmant) [inline deploy/osd-storage/config.yaml:6]: I think these need to be wrapped in double quotes

File: deploy/osd-storage/config.yaml [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+    - key: api.openshift.com/environment
+      operator: In
+      values: [stage]
```
Comment (dustman9000) [inline deploy/osd-storage/config.yaml:9]: Why are we restricting the environment to stage?

File: deploy/osd-storage/cluster-storage-gcp-standard.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: standard
```
Comment (dustman9000) [inline deploy/osd-storage/cluster-storage-gcp-standard.yaml:4]: this should be called `ssd`

Comment (dustman9000) [inline deploy/osd-storage/cluster-storage-gcp-standard.yaml:4]:
```suggestion
  name: ssd
```

File: deploy/osd-storage/cluster-storage-gcp-standard.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+apiVersion: storage.k8s.io/v1
```
Comment (dustman9000) [inline deploy/osd-storage/cluster-storage-gcp-standard.yaml:1]: you can remove this file as we already have this storage class for gcp clusters

File: deploy/osd-storage/cluster-storage-gcp-ssd.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
+apiVersion: storage.k8s.io/v1
+kind: StorageClass
+metadata:
+  name: ssd
+  annotations:
+    storageclass.kubernetes.io/is-default-class: "true"
```
Comment (dustman9000) [inline deploy/osd-storage/cluster-storage-gcp-ssd.yaml:6]: We want this to be false 
```suggestion
    storageclass.kubernetes.io/is-default-class: "false"
```

File: deploy/osd-storage/config.yaml (hunk 2 above)
Comment (dustman9000) [inline deploy/osd-storage/config.yaml:6]:
```suggestion
      values: ["gcp"]
```

File: deploy/osd-storage/config.yaml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+deploymentMode: SelectorSyncSet
```
Comment (dustman9000) [inline deploy/osd-storage/config.yaml:1]: We also want to add CCS to the constraints:
```
  - key: api.openshift.com/ccs
    operator: In
    values: ["true"]
```

File: deploy/osd-storage/config.yaml [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+selectorSyncSet: 
+  matchExpressions:
```
Comment (dustman9000) [inline deploy/osd-storage/config.yaml:3]: We want to only create and not continuously update: https://github.com/openshift/hive/blob/master/pkg/apis/hive/v1/syncset_types.go#L37
```suggestion
  applyBehavior: "CreateOnly"
  matchExpressions: 
```

**Pull request closed without merging**